conda_build/metadata.py

```diff
@@ -451,7 +451,9 @@ def version(self):
         return res
 
     def build_number(self):
-        return int(self.get_value('build/number', 0))
+        number = self.get_value('build/number', 0)
+        # build number can come back as None if no setting (or jinja intermediate)
+        return int(number) if number else 0
```

Comment:
else 0 should be None here, so that build_number triggers jinja invalid messages.